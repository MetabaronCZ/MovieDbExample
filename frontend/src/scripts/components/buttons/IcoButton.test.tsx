import { act } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';
import { fireEvent, getByRole } from '@testing-library/dom';

import { IcoButton } from './IcoButton';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/button/Button', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <IcoButton ico="cross" onClick={() => null} />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    const ico = button.querySelector('svg.fa-xmark');
    expect(ico).toBeInTheDocument();
  });

  it('should be clickable', async () => {
    const onClick = jest.fn();

    const { container } = await waitForComponent(
      <IcoButton ico="cross" onClick={onClick} />,
      { wrapper: TestComponentWrapper },
    );
    expect(onClick).not.toHaveBeenCalled();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    // click on button
    act(() => {
      fireEvent.click(button);
    });
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should not be clickable when disabled', async () => {
    const onClick = jest.fn();

    const { container } = await waitForComponent(
      <IcoButton ico="cross" onClick={onClick} disabled />,
      { wrapper: TestComponentWrapper },
    );
    expect(onClick).not.toHaveBeenCalled();

    const button = getByRole(container, 'button');
    expect(button).toBeInTheDocument();

    // click on button
    act(() => {
      fireEvent.click(button);
    });
    expect(onClick).not.toHaveBeenCalled();
  });
});
