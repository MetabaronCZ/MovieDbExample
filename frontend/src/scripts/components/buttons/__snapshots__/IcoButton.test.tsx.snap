// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/button/Button should render properly 1`] = `
.c0 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c0:disabled {
  cursor: not-allowed;
}

.c1 {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

<div>
  <button
    class="c0 c1"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-xmark fa-lg "
      color="#002d56"
      data-icon="xmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
